x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -135,6 +135,7 @@ public String toString() {
     );
 
     public static final Set<String> ALL_FIELDS = Set.of("*");
+    public static final Set<String> INDEX_METADATA_FIELD = Set.of("_index");
```

Comment:
nit: maybe `INDEX_METADATA_FIELD_SET`? i found it a bit confusing when returning a `..._FIELD` for a return of type `Set`.

Reply:
Everything related to these field names do not have a `set` in their variable names, even if they use `Set` everywhere. And I don't think it should have Set in their name. I regard the Set as an implementation detail to help eliminating duplicates. If I change this to `...._SET` I'd want to change everything related to field names to something ending in `Set` (in IndexResolver, EsqlSession and tests as well). I think it's clear enough like it is now.